app/uk/gov/hmrc/economiccrimelevyaccount/services/ECLAccountService.scala

```diff
@@ -0,0 +1,270 @@
+package uk.gov.hmrc.economiccrimelevyaccount.services
+
+import cats.data.EitherT
+import uk.gov.hmrc.economiccrimelevyaccount.connectors.ECLAccountConnector
+import uk.gov.hmrc.economiccrimelevyaccount.models._
+import uk.gov.hmrc.economiccrimelevyaccount.models.errors.ECLAccountError
+import uk.gov.hmrc.economiccrimelevyaccount.viewmodels.PaymentStatus._
+import uk.gov.hmrc.economiccrimelevyaccount.viewmodels.PaymentType.{Interest, Overpayment, StandardPayment}
+import uk.gov.hmrc.economiccrimelevyaccount.viewmodels._
+import uk.gov.hmrc.http.{HeaderCarrier, UpstreamErrorResponse}
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+import scala.util.control.NonFatal
+
+class ECLAccountService @Inject() (
+  eclAccountConnector: ECLAccountConnector
+)(implicit ec: ExecutionContext) {
+
+  def retrieveFinancialData(implicit
+    hc: HeaderCarrier
+  ): EitherT[Future, ECLAccountError, Option[FinancialData]] =
+    EitherT {
+      eclAccountConnector.getFinancialData.map(Right(_)).recover {
+        case error @ UpstreamErrorResponse(message, code, _, _)
+            if UpstreamErrorResponse.Upstream5xxResponse
+              .unapply(error)
+              .isDefined || UpstreamErrorResponse.Upstream4xxResponse.unapply(error).isDefined =>
+          Left(ECLAccountError.BadGateway(reason = message, code = code))
+        case NonFatal(thr) => Left(ECLAccountError.InternalUnexpectedError(thr.getMessage, Some(thr)))
+      }
+    }
+
+  def retrieveObligationData(implicit
+    hc: HeaderCarrier
+  ): EitherT[Future, ECLAccountError, Option[ObligationData]] =
+    EitherT {
+      eclAccountConnector.getObligationData.map(Right(_)).recover {
+        case error @ UpstreamErrorResponse(message, code, _, _)
+            if UpstreamErrorResponse.Upstream5xxResponse
+              .unapply(error)
+              .isDefined || UpstreamErrorResponse.Upstream4xxResponse.unapply(error).isDefined =>
+          Left(ECLAccountError.BadGateway(reason = message, code = code))
+        case NonFatal(thr) => Left(ECLAccountError.InternalUnexpectedError(thr.getMessage, Some(thr)))
+      }
+    }
+
+  def prepareFinancialDetails(
+    financialDataOption: Option[FinancialData]
+  ): EitherT[Future, ECLAccountError, Option[FinancialViewDetails]] =
+    Try {
+      financialDataOption.flatMap { response =>
+        response.documentDetails.map { documentDetailsList =>
+          val outstandingPayments = documentDetailsList
+            .collect(filterOutstandingPayment)
+            .collect {
+              getOutstandingPayments(documentDetailsList)
+            }
+
+          val paymentsHistory = documentDetailsList
+            .collect(filterOutOverPayment)
+            .flatMap { details =>
+              details.lineItemDetails.map {
+                _.collect(filterOutItemsWithoutClearingReason andThen useOnlyRegularLineItemDetails)
+                  .filter(item => item.isCleared)
```

Comment:
Why not have a partial function so it can be added to the collect statement above?